_inc/lib/admin-pages/class.jetpack-my-jetpack-page.php

```diff
@@ -0,0 +1,181 @@
+
+// Builds the My Jetpack page
+class Jetpack_My_Jetpack_Page extends Jetpack_Admin_Page {
+	// Show the settings page only when Jetpack is connected or in dev mode
+	protected $dont_show_if_not_active = true;
+	function add_page_actions( $hook ) {} // There are no page specific actions to attach to the menu
+
+	// Adds the My Jetpack page, but hides it from the submenu
+	function get_page_hook() {
+		return add_submenu_page( null, __( 'My Jetpack', 'jetpack' ), __( 'My Jetpack', 'jetpack' ), 'jetpack_connect_user', 'my_jetpack', array( $this, 'render' ) );
+	}
+
+	// Renders the view file
+	function page_render() {
+		Jetpack::init()->load_view( 'admin/my-jetpack-page.php' );
+
+		//My Jetpack view tracking, send to MC Stats
+		Jetpack::init()->stat( 'admin', 'my-jetpack' );
+		Jetpack::init()->do_stats( 'server_side' );
+
+	}
+
+	/*
+	 * Handle the change in master user
+	 */
+	function jetpack_my_jetpack_change_user() {
+		if ( ! isset( $_POST['_my_jetpack_nonce'] ) || ! wp_verify_nonce( $_POST['_my_jetpack_nonce'], 'jetpack_change_primary_user' ) ) {
+			wp_die( __( 'Failed permissions, please try again.', 'jetpack' ) );
+			exit;
+		}
+
+		if ( isset( $_POST['jetpack-new-master'] ) ) {
+			$old_master_user   = Jetpack_Options::get_option( 'master_user' );
+			$new_master_user   = $_POST['jetpack-new-master'];
+			$user_token        = Jetpack_Data::get_access_token( $new_master_user );
+			$is_user_connected = $user_token && ! is_wp_error( $user_token );
+			if ( current_user_can( 'manage_options' ) && $is_user_connected ) {
+				Jetpack::log( 'switch_master_user', array( 'old_master' => $old_master_user, 'new_master' => $new_master_user ) );
+				Jetpack_Options::update_option( 'master_user', $new_master_user );
+				Jetpack::state( 'message', 'switch_master' );
+
+				//My Jetpack primary user successfully changed, send to MC Stats
+				Jetpack::init()->stat( 'admin', 'change-primary-successful' );
+				Jetpack::init()->do_stats( 'server_side' );
+
+			}
+		}
+	}
+
+	/*
+	 * Checks to see if there are any other users available to become primary
+	 * Users must both:
+	 * - Be linked to wpcom
+	 * - Be an admin
+	 *
+	 * @return bool
+	 */
+	function jetpack_are_other_users_linked_and_admin() {
+		// If only one admin
+		$all_users = count_users();
+		if ( 2 > $all_users['avail_roles']['administrator'] ) {
+			return false;
+		}
+
+		$users = get_users();
+		$available = array();
+		// If no one else is linked to dotcom
+		foreach ( $users as $user ) {
+			if ( isset( $user->caps['administrator'] ) && Jetpack::is_user_connected( $user->ID ) ) {
+				$available[] = $user->ID;
+			}
+		}
+
+		if ( 2 > count( $available ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/*
+	 * Check if master user has been deleted or is missing somehow
+	 *
+	 * @return bool - True = Master user is still here, False = They're missing
+	 */
+	function jetpack_is_master_user_here() {
+		$master = Jetpack_Options::get_option( 'master_user' );
```

Comment:
I can see that this method is being used in several places, but I don't like duplicating this option query here. Perhaps we can make this method return actual user data if it's there instead of just returning a boolean?